### PR TITLE
add more help info to the CLI

### LINKED ISSUE #585: Edit first line of output in CLI
I just noticed the first line of output in the CLI is this:
`Defang CLI manages services on the Defang cluster`
Can we either delete it or edit it - I don't think we use the term `Defang cluster `anywhere else.

### PR DESCRIPTION
**Pull request by lionello**
Specifically for users that don't know Docker Compose or Dockerfile.

Fixes #585

### CODE REVIEW

File: src/cmd/cli/command/commands.go [hunk 1]
```diff
@@ -829,6 +840,10 @@ var composeUpCmd = &cobra.Command{
 		since := time.Now()
 		deploy, project, err := cli.ComposeUp(cmd.Context(), client, force)
 		if err != nil {
+			if !errors.Is(err, types.ErrComposeFileNotFound) {
+				return err
+			}
+			printDefangHint("To start a new project, do:", "new")
```
[the remark this answers is not in the document]
Comment (lionello) [reply src/cmd/cli/command/commands.go:846]: It's referenced on our samples site:
image

[the remark this answers is not in the document]
Comment (jordanstephens) [reply src/cmd/cli/command/commands.go:845]: It's not described in the help text, what do you think about using `defang generate` in this error message instead? (And maybe in the sample site)

```suggestion
			printDefangHint("To start a new project, do:", "generate")
```

File: src/pkg/cli/new.go [hunk 2]
```diff
@@ -50,7 +50,17 @@ func FetchSamples(ctx context.Context) ([]Sample, error) {
 	return samples, err
 }
 
+// MixinFromSamples copies the sample files into the given directory, skipping existing files.
+func MixinFromSample(ctx context.Context, dir string, name string) error {
+	return copyFromSamples(ctx, dir, []string{name}, true)
+}
```
Comment (jordanstephens) [inline src/pkg/cli/new.go:56]: How does this get used?

Comment (lionello) [reply src/pkg/cli/new.go:56]: The idea here is that we'd publish "mixin samples" that you could "add" to your existing project, for example Dockerfile or compose.yaml files, possible even "frontend" or "backend", etc.. It's related to https://github.com/DefangLabs/samples/pull/165 but I'm still thinking about the dev experience.

File: src/cmd/cli/command/commands.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lionello) [reply src/cmd/cli/command/commands.go:845]: At first we only had GenAI so `defang generate` was born. Later we added the ability to "instantiate" samples and we `defang new` was born. I think it makes sense to separate the two commands, even though one might call the other for backwards compat.

[the remark this answers is not in the document]
Comment (jordanstephens) [reply src/cmd/cli/command/commands.go:845]: That's fine, could we add `new` to the help text then?

File: src/pkg/cli/debug.go [hunk 3]
```diff
@@ -120,6 +120,24 @@ func findMatchingProjectFiles(project *types.Project, services []string) []*defa
 	return files
 }
 
+func IsProjectFile(basename string) bool {
+	return matchAny(patterns, basename)
+}
+
+func matchAny(patterns []string, name string) bool {
```
Comment (nullfunc) [inline src/pkg/cli/debug.go:127]: I would suggest changing the name to matchFilePath or something along those lines. It's not obvious from the name that its not a generic pattern being matched but that of file paths.

File: src/cmd/cli/command/commands.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lionello) [reply src/cmd/cli/command/commands.go:858]: Done

**Review by nullfunc — APPROVED** [review]

**Pull request merged**